ui/src/views/WorkerManager/WMEditWorkerPool/index.jsx

```diff
@@ -0,0 +1,203 @@
+      });
+    }
+  };
+
+  render() {
+    const { isNewWorkerPool, classes } = this.props;
+    const { workerPool, invalidProviderConfig } = this.state;
+
+    return (
+      <Dashboard
+        title={
+          isNewWorkerPool
+            ? 'Worker Manager: Create Worker Type'
+            : 'Worker Manager: Edit Worker Type'
+        }>
+        <TextField
+          label="Enter Worker Type Name..."
+          name="name"
+          error={
+            Boolean(workerPool.name) && !isWorkerTypeNameValid(workerPool.name)
+          }
+          onChange={this.handleInputChange}
+          fullWidth
+          value={workerPool.name}
+          margin="normal"
+        />
+
+        <TextField
+          label="Enter Worker Type Description..."
+          name="description"
+          onChange={this.handleInputChange}
+          fullWidth
+          value={workerPool.description}
+          margin="normal"
+        />
+
+        <TextField
+          label="Enter Owner's Email..."
+          name="owner"
+          error={Boolean(workerPool.owner) && !workerPool.owner.includes('@')}
+          onChange={this.handleInputChange}
+          fullWidth
+          value={workerPool.owner}
+          margin="normal"
+        />
+
+        <FormGroup>
+          <FormControlLabel
+            control={
+              <Switch
+                checked={workerPool.wantsEmail}
+                onChange={this.handleSwitchChange}
+                value="wantsEmail"
+              />
+            }
+            label="Receive emails about errors"
+          />
+        </FormGroup>
+
+        <FormGroup classes={classes.group}>
+          <FormLabel component="provider">Provider:</FormLabel>
+          <TextField
+            id="select-provider-type"
+            select
+            label="Type:"
+            helperText="Which service do you want to run your tasks in?"
+            value={workerPool.providerType}
+            name="providerType"
+            onChange={this.handleInputChange}
+            margin="normal">
+            <MenuItem value="gcp">GCP</MenuItem>
+          </TextField>
+
+          <TextField
+            label="Name:"
+            value={workerPool.providerId}
+            name="providerId"
+            onChange={this.handleInputChange}
+            margin="normal"
+          />
+
+          <List>
+            <ListItem>
+              <FormLabel component="config">Configuration:</FormLabel>
+            </ListItem>
+            <ListItem>
+              <CodeEditor
+                value={JSON.stringify(workerPool.config, null, 2)}
+                onChange={this.handleEditorChange}
+                lint
+              />
+            </ListItem>
+          </List>
+        </FormGroup>
+        <Button
+          spanProps={{ className: classes.createIconSpan }}
+          disabled={invalidProviderConfig}
+          tooltipProps={{ title: 'Save' }}
+          onClick={this.handleCreateWorkerPool}
```

Comment:
yeah, not yet - Coming Soon(TM)

Reply:
No need to implement the logic just yet, but we could have it defined as:

```jsx
handleCreateWorkerPool = () => {};
```

This would avoid having unresolved variables in our code base and have IDE warnings like:

![Screen Shot 2019-05-31 at 2 27 05 PM](https://user-images.githubusercontent.com/3766511/58726629-38169e80-83b0-11e9-9bda-4e686e2b3c24.png)

Reply:
lol ok